app/validators/boolean_validator.rb

```diff
@@ -0,0 +1,14 @@
+# BooleanValidator prevents unexpected Rails type conversion.
+# For example, it prevents the value like ‘foo’ or ‘bar’ from being saved as true.
```

Comment:
考えてみましたが説明が不足しているように思いました。

不足点は

- そもそもAPIの入力値を絞り込みたいこと
- Railsの組み込みバリデーターで実現できないこと

それらの点を追加して次のような説明にしてみました。いかがでしょうか？

In order to clarify the API specification, I would like to restrict the input values to '1', 'true', '0', and 'false' for parameters that receive Boolean values.
However, when using Rails' built-in validator, values like 'foo' and 'bar' are also accepted as true. This is because Rails' built-in validator checks the value after Rails' type casting process.
In Rails' type casting, strings that are not false are converted to true.
The BooleanValidator checks the value before type casting, and prevents values like 'foo' and 'bar' from being saved as true.

また使用例のサンプルもあると、よりイメージしやすくなると思います。
サンプル自体は、まんま `validates :is_public, boolean: true` で良いと思います。

Reply:
ご提案ありがとうございます。
なるほど、概要が伝わるようになってとても良さそうです。

コメントを反映させていただき、使用例を追加しました。
https://github.com/pubannotation/textae-configs/pull/18/commits/c6c14fd7c3154f80fc0596c7062dd6c10e83f47e